<a href="https://colab.research.google.com/github/Nico-GP/Coding-Dojo_Track1/blob/main/Week2/Titanic.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **CODING DOJO**: Titanic
## Creado por: Nicolas Gonzalez
### Descripcion: Calculo de probabilidades de supervivencia en el Titanic
* Creado: 11/01/22
* Ultima Actualizacion: 11/01/22

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd
titanic = pd.read_csv('/content/drive/My Drive/CodingDojo/Week2/titanic.csv')
titanic

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0              1         0       3  ...   7.2500   NaN         S
1              2         1       1  ...  71.2833   C85         C
2              3         1       3  ...   7.9250   NaN         S
3              4         1       1  ...  53.1000  C123         S
4              5         0       3  ...   8.0500   NaN         S
..           ...       ...     ...  ...      ...   ...       ...
886          887         0       2  ...  13.0000   NaN         S
887          888         1       1  ...  30.0000   B42         S
888          889         0       3  ...  23.4500   NaN         S
889          890         1       1  ...  30.0000  C148         C
890          891         0       3  ...   7.7500   NaN         Q

[891 rows x 12 columns]

In [ ]:
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


1. Cabecera de la trama

In [ ]:
titanic.head()

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S
3            4         1       1  ...  53.1000  C123         S
4            5         0       3  ...   8.0500   NaN         S

[5 rows x 12 columns]

2. Porcentaje de supervivencia

In [ ]:
Sup = titanic["Survived"].mean()
Supervivencia = Sup * 100
print(Supervivencia)

38.38383838383838


Sobrevivio el 38,4% de los pasajeros

3. Genero de los supervivientes

In [ ]:
titanic.groupby(["Sex"])[["Survived"]].sum()

Survived
Sex             
female       233
male         109

Sobrevivieron 233 mujeres y 109 hombres

4. Porcentaje de supervivencia con pasaje menor a 10

In [ ]:
Sobrevivientes = titanic["Survived"] == 1
Menos_10 = titanic["Fare"] < 10

Sob_Men_10 = len(titanic[Sobrevivientes & Menos_10]) / len(titanic["Survived"])
Porc_Sob_Men_10 = Sob_Men_10 * 100
Porc_Sob_Men_10

7.519640852974187

7,5% de los sobrevivientes tenian un pasaje menor a 10

5. y 6. Edad promedio de los **NO** sobrevivientes y sobrevivientes

In [ ]:
titanic.groupby(["Survived"])[["Age"]].mean()

Age
Survived           
0         30.626179
1         28.343690

La edad promedio es de:
* 30,6 años para los **no** sobrevivientes
* 28,3 años para los sobrevivientes 

7. Edad promedio de sobrevivientes y no sobrevivientes agrupados por genero

In [ ]:
titanic.groupby(["Sex","Survived"])[["Age"]].mean()

Age
Sex    Survived           
female 0         25.046875
       1         28.847716
male   0         31.618056
       1         27.276022

La edad promedio de:
* Sobrevivientes:
  * Mujeres 28,8 años
  * Hombres 27,3 años
* No Sobrevivientes:
  * Mujeres 25,0 años
  * Hombres 31,6 años